networkx/algorithms/d_separation.py

```diff
@@ -100,6 +153,12 @@ def d_separated(G, x, y, z):
         If any of the input nodes are not found in the graph,
         a :exc:`NodeNotFound` exception is raised.
 
+    Notes
+    -----
+    A d-separating set in a DAG is a set that when conditioned on
+    blocks all paths between the two sets of variables.
```

Comment:
```suggestion
    A d-separating set in a DAG is a set of nodes that
    blocks all paths between the two sets. Nodes in `z`
    block a path if they are part of the path and are not 
    the child in a collider structure or a descendant of a collider.
    A collider structure along a path is ``... -> c <- ...`` where
    ``c`` is the collider node.
```
Is my wording even correct?  I'm adding it to make this doc_string more clear and also a bit more self-contained. But change it as you see fit.

Reply:
Changed the wording slightly